### DISCUSSION TITLE
[FE210] 關於玩家大頭照&實況圖片的問題

### DISCUSSION THREAD

**Discussion by a7474267** [Lidemy/forum#267] [其他問題]
### 作業名稱
4. 從假資料到真資料：Ajax 與 API 串接

### 碰到的問題
1. 成功從API取得玩家大頭照，但大頭照的連結顯示為undefined
2. 成功從API取得實況畫面圖片，但API傳回的資料呈現如下
"https://static-cdn.jtvnw.net/previews-ttv/live_user_rush-{width}x{height}.jpg"
因為沒有定義 {width}x{height}，所以無法直接用 <img src="圖片連結" alt=""></div>
要如何把API傳回的資料，修改為，例如300x300
"https://static-cdn.jtvnw.net/previews-ttv/live_user_rush-300x300.jpg"
然後再渲染到DOM上?

### 預期的行為
有成功取得玩家大頭照，且應該要被渲染在畫面上

### 嘗試過的解決方法
使用了async await，確保資料取得是在非同步，卻能以同步的形式執行，但效果不如預期
玩家大頭照依舊呈現為undefined

### 相關程式碼
https://github.com/a7474267/twitch

**Comment by allenliaooooooo** [comment]
應該就是把拿到的字串做處理就好了，可以到 MDN 看看 string 有哪些 methods 可以使用

**Reply by allenliaooooooo** [reply]
重看了一次發現原來有兩個問題，上面只回答了第二個問題
第一個問題的話，你的 `getUserProfile` 裡呼叫 `axios.get()` 之後，是使用 `.then()` 來接 response
可以再想想 `.then()` 跟 `await` 兩者的差異是什麼
